### PR TITLE
 FEATURE: expandable env keys

### PR DESCRIPTION
**Pull request by OsamaSayegh**
This feature allows you to specify env keys that you want to be able to expand in the env tab so that you can quickly see all of their values without having to navigate through all of the env hashes one by one. Screenshots to clarify what this means:

image

Clicking "4 more" will show you:

image

To configure this feature, do something like this in your app:
```ruby
Logster.config.env_expandable_keys.push(:key, :another_key)
```

### CODE REVIEW

**Comment by SamSaffron** [comment]
@jjaffeux  can you review this?

File: client-app/app/components/env-tab.js [hunk 1]
```diff
@@ -1,27 +1,61 @@
...
 import { computed } from "@ember/object";
 import { buildHashString } from "client-app/lib/utilities";
+import Preload from "client-app/lib/preload";
 
 export default Component.extend({
   current: 1,
 
   didUpdateAttrs() {
     this.set("current", 1);
+    this.set("expanded", null);
```
Comment (jjaffeux) [inline client-app/app/components/env-tab.js]: Group with `setProperties` when possible please

File: client-app/app/components/actions-menu.js [hunk 2]
```diff
@@ -26,6 +26,12 @@ export default Component.extend({
     }
   }),
 
+  willDestroyElement() {
+    this._super(...arguments);
+    const $document = Em.$(document);
+    $document.unbind("click", this.get("bindingFunction"));
```
Comment (jjaffeux) [inline client-app/app/components/actions-menu.js]: why do we do `$document.on("click` and then use unbind and not off ?

File: client-app/app/components/env-tab.js [hunk 3]
```diff
@@ -1,27 +1,61 @@
...
       return buildHashString(this.get("message.env"));
     } else {
-      const currentEnv = this.get("message.env")[this.get("current") - 1];
-      return buildHashString(currentEnv);
+      const currentEnv = Em.$.extend(
+        {},
+        this.get("message.env")[this.get("current") - 1]
+      );
+      const expandableKeys = Preload.get("env_expandable_keys");
+      expandableKeys.forEach(key => {
```
Comment (jjaffeux) [inline client-app/app/components/env-tab.js]: are we sure this is never going to return null or undefind ?

File: client-app/app/components/env-tab.js [hunk 4]
```diff
@@ -1,27 +1,61 @@
...
+          currentEnv[key] = list.length > 1 ? list : list[0];
+        }
+      });
+      return buildHashString(currentEnv, false, this.get("expanded") || []);
     }
   }),
 
+  click(e) {
+    const $elem = Em.$(e.target);
+    const dataKey = $elem.data("key");
```
Comment (jjaffeux) [inline client-app/app/components/env-tab.js]: I usually do `$elem.attr("data-key")` for this reason https://stackoverflow.com/a/8708345

data() is not bad, but I think attr("data-xxx") has a more expected behaviour.

**Comment by jjaffeux** [comment]
I have a limited knowledge of logster, but that looks correct and well tested. Only made minor comments.

**Comment by OsamaSayegh** [comment]
Thanks @jjaffeux for the review, really appreciated! ❤️ I've made those changes and will merge this PR shortly.

**Pull request merged**